### ISSUE TITLE
Convnext image preprocessor raises an AssertionError when comparing logits

### ISSUE THREAD

**Issue by chandrusuresh** [huggingface/transformers#37461]
### System Info

Convnext image preprocessor raises an assert when checking the logits computed from image preprocessor against the expected logits [here](https://github.com/huggingface/transformers/blob/953196a43dae6a3c474165fba7d215fcbc7b7730/src/transformers/models/convnext/convert_convnext_to_pytorch.py#L187).

Are the expected logits correct? What's the source of these values?

Note: I found the bug in the context of working on a PR for #28180 in #37460.

### Who can help?

@NielsRogge 

### Reproduction

To reproduce this, simply run: `python3.12 src/transformers/models/convnext/convert_convnext_to_pytorch.py`  with appropriate arguments and this will raise an AssertionError as described above.

### Expected behavior

All asserts should pass and the code should run without issues.

**Comment by NielsRogge** [comment]
Thanks for the ping, I checked the original repository, where they seem to [use](https://github.com/facebookresearch/ConvNeXt/blob/048efcea897d999aed302f2639b6270aedf8d4c8/main.py#L153) the default ImageNet mean and std.

I just checked, I contributed this one 3 years ago. Here's the [inference script](https://github.com/NielsRogge/ConvNeXt/blob/understanding_convnext/inference.py) I used to obtain logits on the original implementation, and the corresponding [notebook](https://colab.research.google.com/drive/16iFf4eTHLZ0W185nyzEgBQRalIveTOWn?usp=sharing). There I use IMAGENET_DEFAULT_MEAN and IMAGENET_DEFAULT_STD to preprocess the image.

I think the issue was that I only checked whether my model implementation and the original one returned the same logits on the same pixel values, but not necessarily verifying the image processor itself properly. When contributing models at a later time, I always included that in the conversion script, see e.g. [here](https://github.com/huggingface/transformers/blob/953196a43dae6a3c474165fba7d215fcbc7b7730/src/transformers/models/dinov2/convert_dinov2_to_hf.py#L208-L218) for DINOv2. It's a mistake I learned the hard way!

**Comment by chandrusuresh** [comment]
Thanks Niels. I appreciate the info and your response. If you are able to, I'd appreciate your review on #37460. The PR is fixes this bug and partly resolves #28180.
